sysl2/sysl/codegen.go

```diff
@@ -261,19 +282,31 @@ func GenerateCode(
 	if err != nil {
 		return nil, err
 	}
-	if result.GetMap() != nil {
+	switch {
+	case fileNames.GetMap() != nil:
 		filename := fileNames.GetMap().Items["filename"].GetS()
 		logger.Println(filename)
-		r := processRule(g, result, g.Start, logger)
-		codeOutput = append(codeOutput, &CodeGenOutput{filename, r})
-	} else {
+
+		if result.GetMap() != nil {
+			r := processRule(g, result, g.Start, logger)
+			codeOutput = append(codeOutput, &CodeGenOutput{filename, r})
+		} else if result.GetList() != nil {
+			for _, v := range result.GetList().Value {
+				r := processRule(g, v, g.Start, logger)
```

Comment:
Can these 2 lines be refactored out? they are together at least 3 times in this snippet

Reply:
Thanks, I'll have a look.

Reply:
Fixed.